src/core/utils/templating/helpers/sanitizationHelpers.ts

```diff
@@ -0,0 +1,37 @@
+import { SafeString } from "handlebars"
```

Comment:
This looks like too much custom template parsing. Can we use this instead ?
https://handlebarsjs.com/guide/hooks.html#helpermissing

Reply:
- this hook does not handle empty mustaches `{{}}`
- I couldn't find a way to preserve the original curly-braced template format with this hook
- this hook only expects one string in the `value` part. hence it breaks when we have a string like this - `{{ color: "red", display: flex}}`. This would be common in jsx files and would break because of the commas and other special characters

Handle bars is not made in a way to make it possible to preserve undefined template arguments. which is why the custom template parsing seemed like the only way to me.

Reply:
1. Yes. But we can write custom logic for it
2 &3. Try this
```
Handlebars.registerHelper('helperMissing', function() {
  var options = arguments[arguments.length - 1];
  var args = Array.prototype.slice.call(arguments, 0, -1);
  var name = options.name;

  // Check if it's a block helper
  if (typeof options === 'object' && options.hasOwnProperty('fn')) {
    // For block helpers, preserve both opening and closing tags
    return new Handlebars.SafeString(
      '{{#' + name + ' ' + args.join(' ') + '}}' +
      options.fn(this) +
      '{{/' + name + '}}'
    );
  }

  // For non-block helpers, return the original template
  return new Handlebars.SafeString(
    '{{' + name + ' ' + args.join(' ') + '}}'
  );
});
```

Reply:
@wrongsahil I have updated the PR to use [Handlebar's own escape format](https://handlebarsjs.com/guide/expressions.html#escaping-handlebars-expressions) like we discussed